### ISSUE TITLE
Updating a post with HTML in the content/excerpt is buggy.

### ISSUE THREAD

**Issue by picklepete** [WP-API/WP-API#1469]
I've noticed the following oddity when creating then updating a post with HTML in the post content, using WP-API v2.0.

#### Step 1

Create a post.

> POST /wp-json/wp/v2/posts

```json
{
	"content": "<p>Google.</p><p><img src=\"https://www.google.co.uk/images/srpr/logo11w.png\" /></p>", 
	"status": "publish", 
	"title": "Google", 
	"excerpt": ""
}
```

Response (truncated for clarity):

```json
{
    "id": 583,
    "...": "...",
    "content": {
        "raw": "<p>Google.</p><p><img src=\"https://www.google.co.uk/images/srpr/logo11w.png\" /></p>",
        "rendered": "<p>Google.</p>\n<p><img src=\"https://www.google.co.uk/images/srpr/logo11w.png\" /></p>\n"
    }
}
```

So far, so good.

#### Step 2:

Update the post.

> PUT /wp-json/wp/v2/posts/583

```json
{
	"content": "<p>Google's logo.</p><p><img src=\"https://www.google.co.uk/images/srpr/logo11w.png\" /></p>", 
	"status": "publish", 
	"title": "Google's Logo.", 
	"excerpt": ""
}
```

Response (truncated for clarity):

```json
{
    "id": 583,
    "...": "...",
    "content": {
        "raw": "<p>Google\\'s logo.</p><p><img src=\\\"https://www.google.co.uk/images/srpr/logo11w.png\\\" /></p>",
        "rendered": "<p>Google\\&#8217;s logo.</p>\n<p><img src=\\\"https://www.google.co.uk/images/srpr/logo11w.png\\\" /></p>\n"
    }
}
```

#### Step 3:

[Look at your post](http://i.picklepete.co.uk/1439216447.png).

Please correct me if I'm wrong, but I think the cause is: [wp_update_post](https://github.com/WP-API/WP-API/blob/7669432fbe6f6baa1061473c12d2aeb4bc6b9c11/lib/endpoints/class-wp-rest-posts-controller.php#L260) calls [wp_slash](https://github.com/WordPress/WordPress/blob/aace7c2c8d902773f144b0cebdc7a3fda92c8ada/wp-includes/post.php#L3636), when we've previously called `wp_filter_post_kses` [here](https://github.com/WP-API/WP-API/blob/7669432fbe6f6baa1061473c12d2aeb4bc6b9c11/lib/endpoints/class-wp-rest-posts-controller.php#L630) on `post_title`, `post_content` and `post_excerpt`.

I'd love some feedback on this, it's proving to be a blocker for us.

**Comment by danwatt** [comment]
Im seeing this too.

PHP 5.4.36-1
Wordpress 4.3.1
Plugin version 2.0 beta 4

**Comment by danwatt** [comment]
Upgraded PHP, not seeing the issue any more

5.6.14-1

**Comment by kfeoktistoff** [comment]
Hi,
I met the same issue today which Picklepete had reported on. The only difference is that Picklepete used PUT request for updating a post but I used POST as described in the [documentation](http://v2.wp-api.org/reference/posts/):
`POST http://demo.wp-api.org/wp-json/wp/v2/posts/<id>` 

Plugin version Version 2.0-beta6 (and beta5)
PHP 5.6.15-1 (also I downgraded to 5.6.14-1 to fit version from the previous post)
WordPress 4.3.1

**Comment by kfeoktistoff** [comment]
Hi again,
I've downgraded the plugin version to 1.2.4 and changed my REST requests accordingly. Everything works well now.

**Comment by Mange** [comment]
I'd also like a comment on this one. `update` is clearly calling `wp_slash` because it's passed an object[1], but `insert` never does so. Seems like some weird case in WP, but perhaps WP API needs to call it differently to avoid this problem..?

[1]:

```php
function wp_update_post( $postarr = array(), $wp_error = false ) {
  if ( is_object($postarr) ) {
    // Non-escaped post was passed.
    $postarr = get_object_vars($postarr);
    $postarr = wp_slash($postarr);
  }
  // ...
}
```

**Comment by Mange** [comment]
I upgraded from PHP 5.5 to PHP 5.6.4, and the problem still persists.

**Comment by danielbachhuber** [comment]
This was fixed in beta8 / #1772
